### PR TITLE
Make errors more explicit in Secret Pathway

### PR DESCRIPTION
**Pull request by rlisowski**
I added a bunch of `catch` blocks to make beginners life easier. Right now they are getting `UnhandledPromiseRejectionWarning` warning and gets confused.

### CODE REVIEW

File: secret/1_connecting_to_node/connect.js [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 require('dotenv').config();
 
 const main = async () => {
   // Create connection to DataHub Secret Network node
-  const client = new CosmWasmClient(process.env.SECRET_REST_URL)
+  const client = new CosmWasmClient(process.env.SECRET_REST_URL);
 
   // Query chain ID
   const chainId = await client.getChainId()
+    .catch((err) => { console.error('Could not get chain id:\n', err); });
```
Comment (pacmessica) [inline secret/1_connecting_to_node/connect.js:11]:
```suggestion
    .catch((err) => { throw new Error('Could not get chain id: ' + err); });
```

we're logging the error already in the main catch() at the bottom. I think we only really need to throw the error (then it also doesn't try to execute the rest of the code in main)

File: secret/4_transactions/send.js [hunk 2]
```diff
@@ -1,63 +1,65 @@
...
+        amount: '50000',
+        denom: 'uscrt',
+      },
+    ],
+    gas: '100000',
+  };
+
+  const chainId = await client.getChainId()
+    .catch((err) => { console.error('Could not get chain id:\n', err); });
+  const { accountNumber, sequence } = await client.getNonce(accAddress);
```
Comment (pacmessica) [inline secret/4_transactions/send.js:43]: check error here?

File: secret/4_transactions/send.js [hunk 3]
```diff
@@ -1,63 +1,65 @@
...
+  const signBytes = makeSignBytes([sendMsg], fee, chainId, memo, accountNumber, sequence);
+  const signature = await signingPen.sign(signBytes)
+    .catch((err) => { console.error('Could not sign:\n', err); });
+  const signedTx = {
+    msg: [sendMsg],
+    fee,
+    memo,
+    signatures: [signature],
+  };
+  const { logs, transactionHash } = await client.postTx(signedTx);
```
Comment (pacmessica) [inline secret/4_transactions/send.js:53]: catch here?

File: secret/4_transactions/send.js [hunk 4]
```diff
@@ -1,63 +1,65 @@
...
+
+  // Query the tx result
+  const query = { id: transactionHash };
+  const tx = await client.searchTx(query);
```
Comment (pacmessica) [inline secret/4_transactions/send.js:57]: catch here?

**Review by pacmessica — COMMENTED** [review]
I think we should throw the errors, other than that looks alright. 

Your linter is also pretty opinionated 😛 . I feel we should be consistent with the code in the tutorials [here](https://github.com/figment-networks/datahub-learn/tree/master/network-documentation/secret/tutorials)- so using the same linting preferences and also adding these new catch error blocks. I could help with that if you want

**Comment by rlisowski** [comment]
> our linter is also pretty opinionated

yes it is :) https://www.npmjs.com/package/eslint-config-airbnb-base
I can switch to any settings if we have any. Maybe we can add `.eslintrc.js` or smth to this repo.

**Review by pacmessica — APPROVED** [review]
lgtm

there's probably a better place for the eslint file, but think it's ok for now

**Comment by pacmessica** [comment]
accompanying [PR](https://github.com/figment-networks/datahub-learn/pull/8) in learn :)

**Pull request merged**